src/Lecture2.hs

```diff
@@ -154,13 +167,161 @@ You're free to define any helper functions.
 -}
 
 -- some help in the beginning ;)
+
+newtype Health = Health Int
+
+mkHealth :: Int -> Maybe Health
+mkHealth value
+  | value < 0 = Nothing
+  | otherwise = Just $ Health value
+
+unHealth :: Health -> Int
+unHealth (Health health) = health
+
+newtype AttackPower = AttackPower Int
+
+mkAttackPower :: Int -> Maybe AttackPower
+mkAttackPower value
+  | value < 0 = Nothing
+  | otherwise = Just $ AttackPower value
+
+unAttackPower :: AttackPower -> Int
+unAttackPower (AttackPower power) = power
+
+newtype Endurance = Endurance Int
+
+mkEndurance :: Int -> Maybe Endurance
+mkEndurance value
+  | value < 0 = Nothing
+  | otherwise = Just $ Endurance value
+
+unEndurance :: Endurance -> Int
+unEndurance (Endurance endurance) = endurance
+
+data Treasure = Treasure
```

Comment:
I clarified the task a bit to cover cases so dragons would determine the treasure itself, not just "yes" or "no". It will make the implementation a bit more challenging 😅 

But, besides that, the current implementation covers the requirements nicely 👍🏻